packages/components/src/styles/components/dialog-primitive.scss

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// DIALOG PRIMITIVE
+//
+
+
+// MAIN CONTAINER (<dialog>)
+
+// Since the `:where()` selector has a specificity of `0` (see https://developer.mozilla.org/en-US/docs/Web/CSS/:where)
+// the styles declared below are applied to the `.hds-dialog-primitive` element, but are overwritten by any other style applied via classname
+:where(.hds-dialog-primitive) {
+  all: unset; // removes default dialog & inherited styles. Must be the first style to be applied
+  // unsetting all the styles causes it to not inherit the box-sizing from the parent
+  box-sizing: border-box;
+
+  &:not([open]) {
```

Comment:
Why the `&[open]` is declared in the CSS for `Flyout` and `Modal` instead of here? 🤔